### PR TITLE
ORCHESTRATIONS:  Identification Orchestration - Process Identification Requests

### PR DESCRIPTION
**Pull request by MichaelOSullivanAnswer**
CLOSES #AB15802

### CODE REVIEW

File: ISL.Reidentification.Core.Tests.Unit/Services/Orchestrations/Identifications/IdentificationOrchestrationTests.Logic.cs [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    {
+        [Fact]
+        public async Task ShouldCreateAuditAccessRecordIfRequestDoesNotHaveAccessAsync()
+        {
+            // given
+            DateTimeOffset someDateTimeOffset = GetRandomDateTimeOffset();
+            Guid someGuid = Guid.NewGuid();
+            IdentificationRequest someIdentificationRequest = CreateRandomIdentificationRequest(false);
+            IdentificationRequest inputIdentificationRequest = someIdentificationRequest;
+            IdentificationRequest expectedIdentificationRequest = inputIdentificationRequest;
```
Comment (cjdutoit) [inline ISL.Reidentification.Core.Tests.Unit/Services/Orchestrations/Identifications/IdentificationOrchestrationTests.Logic.cs:24]: Dereference to that you can ensure nothing has changed the state.  The test should fail when you do this
```suggestion
            IdentificationRequest expectedIdentificationRequest = inputIdentificationRequest.DeepClone();
```

File: ISL.Reidentification.Core.Tests.Unit/Services/Orchestrations/Identifications/IdentificationOrchestrationTests.Logic.cs [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+            // then
+            returnedIdentificationRequest.Should().BeEquivalentTo(expectedIdentificationRequest);
+
+            this.identifierBrokerMock.Verify(broker =>
+                broker.GetIdentifierAsync(),
+                    Times.Once);
+
+            this.dateTimeBrokerMock.Verify(broker =>
+                broker.GetCurrentDateTimeOffsetAsync(),
+                    Times.Once);
```
Comment (cjdutoit) [inline ISL.Reidentification.Core.Tests.Unit/Services/Orchestrations/Identifications/IdentificationOrchestrationTests.Logic.cs:69]: I was expecting something like this to verify the loop has called the the brokers x times

```cs
            this.dateTimeBrokerMock.Verify(broker =>
                broker.GetCurrentDateTimeOffsetAsync(),
                    Times.Exactly(inputIdentificationRequest.IdentificationItems.Count()));
```

File: ISL.Reidentification.Core.Tests.Unit/Services/Orchestrations/Identifications/IdentificationOrchestrationTests.cs [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+using KellermanSoftware.CompareNetObjects;
+using Moq;
+using Tynamix.ObjectFiller;
+using Xeptions;
+
+namespace ISL.ReIdentification.Core.Tests.Unit.Services.Orchestrations.Identifications
+{
+    public partial class IdentificationOrchestrationTests
+    {
+        private readonly Mock<IReIdentificationService> reIdentificationService;
```
Comment (cjdutoit) [inline ISL.Reidentification.Core.Tests.Unit/Services/Orchestrations/Identifications/IdentificationOrchestrationTests.cs:28]:
```suggestion
        private readonly Mock<IReIdentificationService> reIdentificationServiceMock;
```

File: ISL.Reidentification.Core.Tests.Unit/Services/Orchestrations/Identifications/IdentificationOrchestrationTests.cs [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+        private readonly Mock<IAccessAuditService> accessAuditService;
```
Comment (cjdutoit) [inline ISL.Reidentification.Core.Tests.Unit/Services/Orchestrations/Identifications/IdentificationOrchestrationTests.cs:29]:
```suggestion
        private readonly Mock<IAccessAuditService> accessAuditServiceMock;
```

File: ISL.Reidentification.Core.Tests.Unit/Services/Orchestrations/Identifications/IdentificationOrchestrationTests.Validations.cs [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+
+            this.loggingBrokerMock.Verify(broker =>
+               broker.LogErrorAsync(It.Is(SameExceptionAs(
+                   expectedIdentificationOrchestrationValidationException))),
+                       Times.Once);
+
+            this.accessAuditService.VerifyNoOtherCalls();
+            this.reIdentificationService.VerifyNoOtherCalls();
+            this.dateTimeBrokerMock.VerifyNoOtherCalls();
+            this.loggingBrokerMock.VerifyNoOtherCalls();
```
Comment (cjdutoit) [inline ISL.Reidentification.Core.Tests.Unit/Services/Orchestrations/Identifications/IdentificationOrchestrationTests.Validations.cs:60]: Missing a VerifyNoOtherCalls for one of the dependencies

File: ISL.Reidentification.Core.Tests.Unit/Services/Orchestrations/Identifications/IdentificationOrchestrationTests.Logic.cs [hunk 6]
```diff
@@ -0,0 +1,188 @@
...
+
+            this.reIdentificationService.Verify(service =>
+                service.ProcessReidentificationRequest(It.Is(
+                    SameIdentificationRequestAs(inputHasAccessIdentificationRequest))),
+                        Times.Once);
+
+            this.accessAuditService.VerifyNoOtherCalls();
+            this.reIdentificationService.VerifyNoOtherCalls();
+            this.dateTimeBrokerMock.VerifyNoOtherCalls();
+            this.loggingBrokerMock.VerifyNoOtherCalls();
```
Comment (cjdutoit) [inline ISL.Reidentification.Core.Tests.Unit/Services/Orchestrations/Identifications/IdentificationOrchestrationTests.Logic.cs:185]: Missing a VerifyNoOtherCalls for one of the dependencies

File: ISL.Reidentification.Core.Tests.Unit/Services/Orchestrations/Identifications/IdentificationOrchestrationTests.Exceptions.cs [hunk 7]
```diff
@@ -0,0 +1,176 @@
...
+
+            this.loggingBrokerMock.Verify(broker =>
+               broker.LogErrorAsync(It.Is(SameExceptionAs(
+                   expectedIdentificationOrchestrationDependencyValidationException))),
+                       Times.Once);
+
+            this.accessAuditService.VerifyNoOtherCalls();
+            this.reIdentificationService.VerifyNoOtherCalls();
+            this.dateTimeBrokerMock.VerifyNoOtherCalls();
+            this.loggingBrokerMock.VerifyNoOtherCalls();
```
Comment (cjdutoit) [inline ISL.Reidentification.Core.Tests.Unit/Services/Orchestrations/Identifications/IdentificationOrchestrationTests.Exceptions.cs:64]: Missing a VerifyNoOtherCalls for one of the dependencies

File: ISL.Reidentification.Core.Tests.Unit/Services/Orchestrations/Identifications/IdentificationOrchestrationTests.Exceptions.cs [hunk 8]
```diff
@@ -0,0 +1,176 @@
...
+
+            this.loggingBrokerMock.Verify(broker =>
+               broker.LogErrorAsync(It.Is(SameExceptionAs(
+                   expectedIdentificationOrchestrationDependencyException))),
+                       Times.Once);
+
+            this.accessAuditService.VerifyNoOtherCalls();
+            this.reIdentificationService.VerifyNoOtherCalls();
+            this.dateTimeBrokerMock.VerifyNoOtherCalls();
+            this.loggingBrokerMock.VerifyNoOtherCalls();
```
Comment (cjdutoit) [inline ISL.Reidentification.Core.Tests.Unit/Services/Orchestrations/Identifications/IdentificationOrchestrationTests.Exceptions.cs:114]: Missing a VerifyNoOtherCalls for one of the dependencies

File: ISL.Reidentification.Core.Tests.Unit/Services/Orchestrations/Identifications/IdentificationOrchestrationTests.Exceptions.cs [hunk 9]
```diff
@@ -0,0 +1,176 @@
...
+
+            this.loggingBrokerMock.Verify(broker =>
+               broker.LogErrorAsync(It.Is(SameExceptionAs(
+                   expectedIdentificationOrchestrationServiceException))),
+                       Times.Once);
+
+            this.accessAuditService.VerifyNoOtherCalls();
+            this.reIdentificationService.VerifyNoOtherCalls();
+            this.dateTimeBrokerMock.VerifyNoOtherCalls();
+            this.loggingBrokerMock.VerifyNoOtherCalls();
```
Comment (cjdutoit) [inline ISL.Reidentification.Core.Tests.Unit/Services/Orchestrations/Identifications/IdentificationOrchestrationTests.Exceptions.cs:173]: Missing a VerifyNoOtherCalls for one of the dependencies

**Review by cjdutoit — APPROVED** [review]

File: ISL.Reidentification.Core.Tests.Unit/Services/Orchestrations/Identifications/IdentificationOrchestrationTests.Validations.cs (hunk 5 above)
Comment (MichaelOSullivanAnswer) [reply ISL.Reidentification.Core.Tests.Unit/Services/Orchestrations/Identifications/IdentificationOrchestrationTests.Validations.cs:60]: this has been updated

File: ISL.Reidentification.Core.Tests.Unit/Services/Orchestrations/Identifications/IdentificationOrchestrationTests.Logic.cs (hunk 6 above)
Comment (MichaelOSullivanAnswer) [reply ISL.Reidentification.Core.Tests.Unit/Services/Orchestrations/Identifications/IdentificationOrchestrationTests.Logic.cs:186]: this has been updated

**Pull request merged**